airbyte-ci/connectors/pipelines/pipelines/actions/environments.py

```diff
@@ -1009,11 +1009,47 @@ def with_crane(
     return base_container
 
 
-def mounted_connector_secrets(context: PipelineContext, secret_directory_path="secrets") -> Callable:
-    def mounted_connector_secrets_inner(container: Container):
+async def mounted_connector_secrets(context: PipelineContext, secret_directory_path: str) -> Callable[[Container], Container]:
+    # By default, mount the secrets properly as dagger secret files.
+    #
+    # This will cause the contents of these files to be scrubbed from the logs. This scrubbing comes at the cost of
+    # unavoidable latency in the log output, see next paragraph for details as to why. This is fine in a CI environment
+    # however this becomes a nuisance locally: the developer wants the logs to be displayed to them in an as timely
+    # manner as possible. Since the secrets aren't really secret in that case anyway, we mount them in the container as
+    # regular files instead.
+    #
+    # The buffering behavior that comes into play when logs are scrubbed is both unavoidable and not configurable.
+    # It's fundamentally unavoidable because dagger needs to match a bunch of regexes (one per secret) and therefore
+    # needs to buffer at least as many bytes as the longest of all possible matches. Still, this isn't that long in
+    # practice in our case. The real problem is that the buffering is not configurable: dagger relies on a golang
+    # library called transform [1] to perform the regexp matching on a stream and this library hard-codes a buffer
+    # size of 4096 bytes for each regex [2].
```

Comment:
🤯

Reply:
I know, right... Go is full of this kind of mickey-mouse crap. 

FYI see also https://github.com/airbytehq/airbyte/issues/30394 which tracks the remaining work around this issue, essentially to ask the dagger maintainers to do something.